.github/workflows/compile.yml

```diff
@@ -0,0 +1,106 @@
+          sudo apt-get update ; sudo apt-get install gcc-aarch64-linux-gnu binutils-aarch64-linux-gnu gcc-aarch64-linux-gnu
+      
+      - name: Compile
+        env:
+          CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER: /usr/bin/aarch64-linux-gnu-gcc
+        run: |
+          cargo build --release --target=aarch64-unknown-linux-gnu
+
+      - name: Copy around
+        run: | 
+          mkdir release
+          cp target/aarch64-unknown-linux-gnu/release/hanachan release/hanachan_linux_aarch64 || /bin/true
+
+      - name: Upload distribution
+        uses: actions/upload-artifact@v2
+        with:
+          name: release
+          path: | 
+            release
+
+
+  build-linux-windows:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install
+        run: |
+          rustup target add i686-unknown-linux-gnu
+          rustup target add x86_64-unknown-linux-gnu
+          #
+          rustup target add i686-pc-windows-gnu
+          rustup target add x86_64-pc-windows-gnu
+          sudo apt-get update ; sudo apt-get install mingw-w64 gcc-multilib
+      
+      - name: Compile
+        run: |
+          cargo build --release --target=i686-unknown-linux-gnu
+          cargo build --release --target=x86_64-unknown-linux-gnu
+          #
+          RUSTFLAGS="-C panic=abort -C embed-bitcode=yes -C lto" cargo build --release --target=i686-pc-windows-gnu
+          cargo build --release --target=x86_64-pc-windows-gnu
+
+      - name: Copy around
+        run: | 
+          mkdir release
+          cp target/i686-unknown-linux-gnu/release/hanachan release/hanachan_linux_i686 || /bin/true
+          cp target/x86_64-unknown-linux-gnu/release/hanachan release/hanachan_linux_x86_64 || /bin/true
+          #
+          cp target/x86_64-pc-windows-gnu/release/hanachan.exe release/hanachan_windows64.exe || /bin/true
+          cp target/i686-pc-windows-gnu/release/hanachan.exe release/hanachan_windows32.exe || /bin/true
+
+      - name: Upload distribution
+        uses: actions/upload-artifact@v2
+        with:
+          name: release
+          path: | 
+            release
+
+  build-osx:
+    runs-on: macos-latest
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install
+        run: |
+          rustup target add x86_64-apple-darwin
+          # rustup target add i686-apple-darwin
+          # rustup target add aarch64-apple-darwin
+      
+      - name: Compile
+        run: |
+          cargo build --release --target=x86_64-apple-darwin
+          # cargo build --release --target=i686-apple-darwin
+          # cargo build --release --target=aarch64-apple-darwin
+
+      - name: Copy around
+        run: | 
+          mkdir release
+          cp target/x86_64-apple-darwin/release/hanachan release/hanachan_mac64 || /bin/true
+          # cp target/i686-apple-darwin/release/hanachan release/hanachan_mac32 || /bin/true
```

Comment:
10.6 was the latest version to support 32-bit macs and support for it was dropped in 2014, so this can safely be removed.